### PR TITLE
Add password complexity feature

### PR DESCRIPTION
**Pull request by CPatchane**
* Add password complexity helper + feature.
* Fix also bug breadcrumb display in password component and add password input placeholder.
* Styles Improvements.

### CODE REVIEW

File: client/app/views/steps/password.coffee [hunk 1]
```diff
@@ -32,6 +36,29 @@ module.exports = class PasswordView extends StepView
         @$visibilityIcon.attr 'xlink:href', data.visibilityIcon
 
 
+    initialize: (args...) ->
+        super args...
+        # lowest level is 1 to display a red little part
+        @passwordStrength = {percentage: 1, label: 'weak'}
```
Comment (gregorylegarec) [inline client/app/views/steps/password.coffee]: Shouldn't the strenght be given by a call to passwordHelper, to be sure to have the same logic as the one implemented in the helper ?

File: client/app/views/steps/password.coffee [hunk 2]
```diff
@@ -32,6 +36,29 @@ module.exports = class PasswordView extends StepView
...
+        @updatePasswordStrength = updatePasswordStrength.bind(@)
+
+
+    updatePasswordStrength= _.throttle( ->
+        password = @$('input[name=password]').val()
```
Comment (gregorylegarec) [inline client/app/views/steps/password.coffee]: We should use to cache jQuery queries, as every call is pretty expensive. (See http://gregfranko.com/blog/jquery-best-practices/). We can to something like
```coffeescript
@passwordInput ?= @$('input[name=password')
```
and then work only with `@passwordInput`.

File: client/app/views/steps/password.coffee [hunk 3]
```diff
@@ -32,6 +36,29 @@ module.exports = class PasswordView extends StepView
...
+        @passwordStrength = passwordHelper.getStrength password
+
+        if @passwordStrength.percentage is 0
+            @passwordStrength.percentage = 1
+        @$('progress').attr 'value', @passwordStrength.percentage
+        @$('progress').attr 'class', 'pw-' + @passwordStrength.label
+        @$inputPassword.removeClass('error')
```
Comment (gregorylegarec) [inline client/app/views/steps/password.coffee]: Same remarks about jQuery queries caching.

File: client/app/views/steps/password.coffee (hunk 2 above)
Comment (CPatchane) [inline client/app/views/steps/password.coffee]: Good point! 👍

**Comment by CPatchane** [comment]
@gregorylegarec: For the two first points, I just pushed the transfix sources and translated them, that should be fixed.

For the last point, I think the password is too short it's why it's too weak. The strenght is also computed according to the password length. But you can also review the password strength helper here in order to be sure.

File: client/app/views/steps/password.coffee (hunk 1 above)
Comment (CPatchane) [inline client/app/views/steps/password.coffee]: You're right, it's more maintainable. I'll fix it.

**Comment by gregorylegarec** [comment]
Thanks @CPatchane 👍

**Pull request merged**